JupyterHub/Vagrantfile

```diff
@@ -18,42 +18,55 @@ Vagrant.configure("2") do |config|
         node.vm.hostname = "jupyterhub"
         node.disksize.size = "50GB"
         node.vm.synced_folder "Shared", "/vagrant_shared"
-        node.vm.synced_folder "../common", "/vagrant_common"
         node.vm.network "public_network"
         node.vm.network "private_network", ip: "10.0.0.40", virtualbox__intnet: true
         node.vm.provision :shell, inline: <<-SHELL
         sudo su
         cd /root
 
         yum -y install https://centos7.iuscommunity.org/ius-release.rpm
-        yum -y install epel-release python36u python36u-dev python36u-devel npm wget net-tools krb5-libs krb5-workstation git krb5-devel
+        yum -y install epel-release python36u python36u-dev python36u-devel npm wget net-tools krb5-libs krb5-workstation git krb5-devel java-1.8.0-openjdk nginx
         yum -y install python36u-pip
 
-        ln -s /bin/pip3.6 /bin/pip
-        pip install --upgrade pip
-        pip install jupyterhub
+        setsebool -P httpd_can_network_connect 1
+
+        version=2.1.3
+
+        python3 -m venv /opt/jupyterhub/
+        /opt/jupyterhub/bin/python3 -m pip install --upgrade pip
+        /opt/jupyterhub/bin/python3 -m pip install wheel
+        /opt/jupyterhub/bin/python3 -m pip install jupyterhub jupyterlab
         npm install -g configurable-http-proxy
-        pip install notebook
-        pip install kerberos
-        pip install shyaml
 
-        cp /vagrant_shared/hosts /etc/hosts
-        ulimit -n 10000
-        echo 1 > /proc/sys/vm/swappiness
-        printf "\nvm.swappiness=1\n" >> /etc/sysctl.conf
+        mkdir -p /opt/jupyterhub/etc/jupyterhub/
+        \\cp /vagrant_shared/jupyterhub_config.py /opt/jupyterhub/etc/jupyterhub/
+        \\cp /vagrant_shared/jupyterhub.service /etc/systemd/system/jupyterhub.service
 
-        chmod +x /vagrant_*/*.sh
+        rm -f /etc/nginx/nginx.conf
+        \\cp /vagrant_shared/nginx.conf /etc/nginx/nginx.conf
+        systemctl restart nginx
 
-        /vagrant_common/setupAuthentication.sh
+        systemctl daemon-reload
+        systemctl enable jupyterhub.service
+        systemctl start jupyterhub.service
 
-        cd /tmp
-        \\cp /vagrant_shared/auth.py /usr/lib/python3.6/site-packages/jupyterhub/auth.py
-        \\cp /vagrant_shared/jupyterhub_config.py .
+        chmod +x /vagrant_*/*.sh
 
         useradd user
         echo "user:userpw" | chpasswd
-        
-        ifconfig | grep "192.168"
+
+        mkdir -p /usr/local/bin/apache-spark
+        wget https://archive.apache.org/dist/spark/spark-$version/spark-$version-bin-hadoop2.7.tgz
+        tar -xzvf spark-$version-bin-hadoop2.7.tgz
+        mv spark-$version-bin-hadoop2.7/* /usr/local/bin/apache-spark
+
+        /opt/jupyterhub/bin/python3 -m pip install --upgrade toree
+        /opt/jupyterhub/bin/jupyter toree install --spark_home=/usr/local/bin/apache-spark/ --interpreters=Scala,SQL

```

Comment:
Python?

Reply:
So apparently Toree has removed support for PySpark and SparkR, based on the theory that they are already supported well in Jupyter through IPython and the R kernel. See https://www.mail-archive.com/dev@toree.incubator.apache.org/msg01865.html for the relevant email chain.